### PR TITLE
refactor: reformat, versioning and minor fixes

### PR DESCRIPTION
**Pull request by ShookLyngs**
## Changes
- Added prettier/husky/lint-staged to reformat code automatically
- Added changests as the version manager
- Fixed an Omnilock flag checking issue
- Added a release ci to publish updates
- Reformatted/Simplified code

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -53,7 +43,13 @@ export function isAnyoneCanPay(script: Script | undefined, config: SporeConfig<s
     return false;
   }
   if (isOmnilockScript(script, config)) {
-    return script.args.slice(44, 46) === '02';
+    const flag = number.Uint8.unpack(`0x${script.args.slice(44, 46)}`);
```
Comment (ashuralyk) [inline src/utils.ts:48]: I guess here needs more information for a better explanation

**Comment by ashuralyk** [comment]
@ShookLyngs after this PR merged, Philosopher Stone won't do anything on it?

**Comment by ShookLyngs** [comment]
> @ShookLyngs after this PR merged, Philosopher Stone won't do anything on it?

No breaking changes, just normal updates.

File: src/utils.ts (hunk 1 above)
Comment (ShookLyngs) [reply src/utils.ts:48]: Added comment and reference

**Review by Dawn-githup — APPROVED** [review]

**Pull request merged**